worker/src/ocall_bridge/bridge_api.rs

```diff
@@ -0,0 +1,80 @@
+/*
+    Copyright 2019 Supercomputing Systems AG
+    Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+use crate::ocall_bridge::component_factory::OCallBridgeComponentFactory;
+use lazy_static::lazy_static;
+use log::*;
+use parking_lot::RwLock;
+use sgx_types::{
+    sgx_epid_group_id_t, sgx_platform_info_t, sgx_quote_nonce_t, sgx_quote_sign_type_t,
+    sgx_report_t, sgx_spid_t, sgx_status_t, sgx_target_info_t, sgx_update_info_bit_t,
+};
+use std::sync::Arc;
+use std::vec::Vec;
+
+#[cfg(test)]
+use mockall::predicate::*;
+#[cfg(test)]
+use mockall::*;
+
+lazy_static! {
+    static ref COMPONENT_FACTORY: RwLock<Option<Arc<dyn OCallBridgeComponentFactory + Send + Sync>>> =
+        RwLock::new(None);
+}
+
+pub struct Bridge {}
+
+impl Bridge {
+    pub fn get_ra_api() -> Arc<dyn RemoteAttestationOCall> {
+        debug!("Requesting RemoteAttestation OCall API instance");
+
+        COMPONENT_FACTORY
+            .read()
+            .as_ref()
+            .expect("Component factory has not been set. Use `initialize()`")
+            .get_ra_api()
+    }
+
+    pub fn initialize(component_factory: Arc<dyn OCallBridgeComponentFactory + Send + Sync>) {
+        debug!("Initializing OCall bridge with component factory");
+
+        *COMPONENT_FACTORY.write() = Some(component_factory);
+    }
+}
+
+#[cfg_attr(test, automock)]
+pub trait RemoteAttestationOCall {
```

Comment:
Trait for the RA o-call, using attributes from the mocking framework

Reply:
Ah nice, while checking the mock libs, I also deemed automock the most promising.

Reply:
So the library I used, is https://crates.io/crates/mockall , not `automock`. That is just the attribute you set on the trait..

Reply:
ah sorry, but it is still the one i meant.